app/controllers/tests_controller.rb

```diff
@@ -0,0 +1,151 @@
+class TestsController < ApplicationController
+  before_action :require_login
+  before_action :load_test, only: [:update, :show, :edit]
+  before_action :has_authorization_with_test?, only: [:show]
+
+  def create
+    @test = current_user.tests.build(test_params)
+    if @test.save
+      add_questions_to_test
+      redirect_to edit_test_path @test
+    else
+      flash.now[:danger] = t "tests.errors.questions_fail"
+      @subject = Subject.find_by id: params[:subject_id]
```

Comment:
```suggestion
      @subject = @test.subject
```
chị nghĩ ri nhỉ chứ cần j load lại subject
mà load lại mà find_by ko check subject thì ko ổn